server/services/supabase_client.py

```
@@ -0,0 +1,20 @@
+def get_supabase_client() -> Client:
+    """Return a cached Supabase client using service_role key.
+
+    Uses service_role to bypass RLS — this runs server-side in a trusted
+    context. User scoping is enforced via WHERE clauses, not RLS.
+    """
+    url = os.environ.get("SUPABASE_URL")
+    key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")
+    if not url or not key:
+        raise RuntimeError(
+            "SUPABASE_URL and SUPABASE_SERVICE_ROLE_KEY must be set in environment."
+        )
```

Ruff (0.15.5)

[warning] Avoid specifying long messages outside the exception class

(TRY003)